generator/internal/task_processor.go

```diff
@@ -137,6 +145,10 @@ func (p *taskProcessor) completeTask(logger lager.Logger, container executor.Con
 			p.failTask(logger, container.Guid, TaskCompletionReasonInvalidTransition)
 		}
 		return
+	} else if container.RunResult.Failed {
+		// When we have done BBS.CompleteTask on a failed Task increment the counter
+		// It will be incremented on every call to FailTask in the p.failTask() method
+		TasksFailed.Increment()
```

Comment:
As a cleanup note, the 'else' part here is unnecessary. An 'if' by itself will suffice.

But as a more substantial question, we're not sure why CompleteTask needs to be called before checking the container.RunResult.Failed. Is it possible for the CompleteTask method itself to set `RunResult.Failed` from True to False? If that is possible, then we would get a `TaskSucceeded` and a `TaskFailed` metric for the same task, which seems strange.

On the other hand, if it's not possible for the RunResult.Failed to be changed by the CompleteTask call to the BBS, then this check on line 148 seems to belong on line 136 instead, to keep the code more readable.